### ISSUE TITLE
RKE deletes user addons on reconcile and redeploys before rke up 

### ISSUE THREAD

**Issue by kinarashah** [rancher/rancher#38749]
**Issue** 

A change was introduced in RKE v1.3.10-v1.3.13 and Rancher v2.6.7 where RKE deletes the user addons on rke up and then redeploys them. This was to fix an issue where user addon changes were lost after cluster upgrades. [#36985](https://github.com/rancher/rancher/issues/36985)

This is an issue because user addon templates could include namespaces. For instance, older hardened rancher clusters have `cattle-system` namespace in the user addons per https://rancher.com/docs/rancher/v2.0-v2.4/en/security/rancher-2.4/hardening-2.4/#reference-hardened-rke-template-configuration. Cluster agents run in the cattle-system namespace and RKE can no longer talk to the cluster with dialer issues so the redeploy doesn't work.

**Steps to Reproduce**
- Create a downstream RKE cluster 
- Edit the cluster.yml to have `cattle-system` in the user addons 
```
rancher_kubernetes_engine_config:
[...]
  addon_job_timeout: 30
  addons: |-
    ---
    apiVersion: v1
    kind: Namespace
    metadata:
      name: cattle-system
[...]
```
- Wait for the cluster to update and come up active 
- Upgrade the cluster by changing k8s version or extra_args for any k8s component 
- Cluster is stuck in upgrading with dialer errors, cluster agent is deleted on the node 
```
Building Kubernetes cluster[INFO ] [dialer] Setup tunnel for host [167.172.178.66][INFO ] [dialer] Setup tunnel for host [64.225.99.151][ERROR]
 Failed to set up SSH tunneling for host [64.225.99.151]: Can't retrieve
 Docker Info: error during connect: Get 
"http://%2Fvar%2Frun%2Fdocker.sock/v1.24/info": can not build dialer to 
[c-c9lt7:m-7e40fa88cf5f][INFO ] [dialer] Setup tunnel for host [68.183.220.116][ERROR]
 Failed to set up SSH tunneling for host [68.183.220.116]: Can't 
retrieve Docker Info: error during connect: Get 
"http://%2Fvar%2Frun%2Fdocker.sock/v1.24/info": can not build dialer to 
[c-c9lt7:m-bd97bd0733d0][ERROR]
```

**Workaround before upgrade** 
- Remove `cattle-*` and other user namespaces from the user addon template before upgrading the cluster using RKE v1.3.10-v1.3.13 and Rancher v2.6.7. 

**Recover after upgrade** 
We could redeploy the agent manually and try to recover the cluster, this worked for a few clusters upgraded in Rancher with `cattle-system` in the user addon template. There could be other issues which will need to be dealt with individually depending on the cluster configuration. 

- Redeploy cluster-agent manually using https://github.com/rancherlabs/support-tools/tree/master/cluster-agent-tool, cluster-agent pods are running but cluster stays in Disconnected stage 
- Update Authorize Cluster Endpoint to false and trigger a cluster reconciliation 

SURE-5146

**Comment by markusewalker** [comment]
**TEST PLAN**
1. Setup Rancher, create a standard user and login as that user.
2. Provision a downstream RKE1 node driver cluster to the second highest K8s version available.
     - While provisioning, edit the cluster's YAML and add the following underneath the `rancher_kubernetes_engine_config` block:
```
addon_job_timeout: 30
addons: |-
  ---
  apiVersion: v1
  kind: Namespace
  metadata:
    name: cattle-system
```
3. Validate that the cluster and nodes appear in a state of `Active` in Cluster Management and Cluster Explorer.
4. Edit the cluster's config and validate that `cattle-system` is still in the addons as added in step 2.
5. Upgrade the cluster's K8s version to the highest K8s version available.
6. Validate if the cluster is able to successfully upgrade and the cluster and nodes are in a state of `Active`.

**Comment by markusewalker** [comment]
Reproduced this issue on `v2.6.7`.

**ENVIRONMENT DETAILS**
- Rancher install: Docker
- Rancher version
     - Reproduced: `v2.6.7`
- Browser: Chrome

**TEST RESULT**
REPRODUCED

**REPRODUCTION STEPS**
1. Setup Rancher, created a standard user and login as that user.
2. Provisioned a downstream RKE1 node driver cluster to the second highest K8s version available.
     - While provisioning, edited the cluster's YAML and add the following underneath the `rancher_kubernetes_engine_config` block:
```
addon_job_timeout: 30
addons: |-
  ---
  apiVersion: v1
  kind: Namespace
  metadata:
    name: cattle-system
```
3. Validated that the cluster and nodes appear in a state of `Active` in Cluster Management and Cluster Explorer.
4. Edited the cluster's config and validate that `cattle-system` is still in the addons as added in step 2.
5. Upgraded the cluster's K8s version to the highest K8s version available.
6. During provisioning, monitored the cluster's `Provisioning Log` and validated that I am seeing the reported issue:
```
[INFO ] [addons] Setting up user addons	
11:02:37 am	[INFO ] [addons] Removing old user addons
11:08:51 am	[ERROR] Failed to deploy addon execute job [rke-user-includes-addons]: Failed to get job complete status for job rke-user-addon-delete-job in namespace kube-system: Get "https://...:6443/apis/batch/v1/namespaces/kube-system/jobs/rke-user-addon-delete-job?timeout=30s": cluster agent disconnected
11:08:51 am	[INFO ] Finished building Kubernetes cluster successfully
11:11:22 am	[INFO ] Initiating Kubernetes cluster
11:11:22 am	[INFO ] Successfully Deployed state file at [management-state/rke/rke-2186161090/cluster.rkestate]
11:11:22 am	[INFO ] Building Kubernetes cluster
11:11:22 am	[INFO ] [dialer] Setup tunnel for host […]
11:11:22 am	[INFO ] [dialer] Setup tunnel for host […]
11:11:22 am	[ERROR] Failed to set up SSH tunneling for host […]: Can't retrieve Docker Info: error during connect: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/info": can not build dialer to [c-2k7kh:m-kcbb8]
11:11:22 am	[ERROR] Failed to set up SSH tunneling for host […]: Can't retrieve Docker Info: error during connect: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/info": can not build dialer to [c-2k7kh:m-g4rsk]
11:11:22 am	[ERROR] Removing host [173.230.151.169] from node lists
11:11:22 am	[ERROR] Removing host [173.230.151.233] from node lists
11:12:43 am	[INFO ] Initiating Kubernetes cluster
11:12:43 am	[INFO ] Successfully Deployed state file at [management-state/rke/rke-539469822/cluster.rkestate]
11:12:43 am	[INFO ] Building Kubernetes cluster
11:12:43 am	[INFO ] [dialer] Setup tunnel for host […]
11:12:43 am	[INFO ] [dialer] Setup tunnel for host […]
11:12:43 am	[ERROR] Failed to set up SSH tunneling for host […]: Can't retrieve Docker Info: error during connect: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/info": can not build dialer to [c-2k7kh:m-kcbb8]
11:12:43 am	[ERROR] Failed to set up SSH tunneling for host […]: Can't retrieve Docker Info: error during connect: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/info": can not build dialer to [c-2k7kh:m-g4rsk]
11:12:43 am	[ERROR] Removing host […] from node lists
11:12:43 am	[ERROR] Removing host […] from node lists
```

**Comment by BBQigniter** [comment]
This is really strange - something must be different between RKE 1.23.8 and 1.23.10

We have here Rancher v2.6.6 running which currently manages multiple RKE clusters deployed on Docker. A few clusters have already version 1.23.8 where we did not see the upgrade failing from 1.22.10 in the past. 

In the last days I tried to do the 1.23 upgrade on two other clusters from 1.22.10. Unfortunately 1.23.8 is not in the dropdown anymore to choose from - so I tried 1.23.10. The first cluster had unfortunately only one master and 3 worker-nodes and the cluster ended then in some state where we had to completely recreate it (now with 3 master/worker nodes + 1 worker-only running with 1.23.10). 

Today I tried another 3-node cluster where each node is a master and worker.

On the first try we saw the error `Failed to set up SSH tunneling for host` in the provision logs only occuring with one node and the upgrade did not proceed anymore. So we recovered via snapshot. Rancher did some magic and somehow recovered after the whole cluster was unresponsive for some time. After it got back as "available" the previously failing node was suddenly removed from the nodes list but not from the "Machines" list in the "Cluster Management". I deleted it from the list, did a cleanup and reregistered it. The cluster now was back as it should be.

So next I did on all nodes CentOS updates (`kernel-ml` and `containerd` was updated), rebooted the nodes one-by-one, waited until cluster was back to normal. Had another look at this github-issue, checked the cluster-config and made sure that there is no "addons" section in there. Tried the upgrade again, after creating a snapshot.
This time suddenly 2 out of 3 nodes showed the `Failed to set up SSH tunneling for host` in the provision logs. Again reverted via snapshot, this time the two affected nodes got removed from the nodes list, cleaned up nodes, and readded. Cluster is again thankfully back to normal with version 1.22.10.

In the github-issue here, also `extra_args` is mentioned to probably cause the problem.

> * Upgrade the cluster by changing k8s version or extra_args for any k8s component

In the config of a cluster I can only find `extra_args` in the `services` part for `etcd`:

```
  services:
    etcd:
      ...
      extra_args:
        election-timeout: '5000'
        heartbeat-interval: '500'
```

But as far as I know - those are default, as I can see these settings on every cluster. We only have defined some other settings for `audit_log`-configurations.

So i assume we only can/should wait for 2.6.9 until this issue is fixed?


BTW - In our LAB-environment we have Rancher 2.6.8 which manages one lab-cluster where we did not see such upgrade issues. I think, there we too upgraded from 1.22.10 to 1.23.8 and later up to 1.24.4 without problems.

**Comment by kinarashah** [comment]
@BBQigniter Since you don't have `addons:` configured for the cluster, I am wondering if you ran into some other issue especially because the reconciling helped recover the cluster but it would've failed constantly with this issue. Did you also see dialer error msg followed by `Failed to set up SSH tunneling for host` in the logs, could you share the full error msg?

**Comment by BBQigniter** [comment]
@kinarashah - here are the logs - I hope they are complete as I had found them in some binary etcd-snapshot :o as Rancher seems not to write those logs just in some file.

This is the log from the first try, where only one node failed (IPs are obfuscated):

```
2022-09-12T07:03:56Z [INFO ] [dialer] Setup tunnel for host [123.123.123.49]
2022-09-12T07:03:56Z [INFO ] [dialer] Setup tunnel for host [123.123.123.47]
2022-09-12T07:03:56Z [INFO ] [dialer] Setup tunnel for host [123.123.123.48]
2022-09-12T07:03:56Z [ERROR] Failed to set up SSH tunneling for host [123.123.123.47]: Can't retrieve Docker Info: error during connect: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/info": can not build dialer to [c-7xwnb:m-5409343b1120]
2022-09-12T07:03:56Z [ERROR] Removing host [123.123.123.47] from node lists
2022-09-12T07:03:56Z [INFO ] [network] No hosts added existing cluster, skipping port check
2022-09-12T07:03:56Z [INFO ] [certificates] Deploying kubernetes certificates to Cluster nodes
2022-09-12T07:04:02Z [INFO ] [reconcile] Rebuilding and updating local kube config
2022-09-12T07:04:02Z [INFO ] Successfully Deployed local admin kubeconfig at [management-state/rke/rke-3162177412/kube_config_cluster.yml]
2022-09-12T07:04:02Z [INFO ] [reconcile] host [123.123.123.48] is a control plane node with reachable Kubernetes API endpoint in the cluster
2022-09-12T07:04:02Z [INFO ] [certificates] Successfully deployed kubernetes certificates to Cluster nodes
2022-09-12T07:04:02Z [INFO ] [file-deploy] Deploying file [/etc/kubernetes/kube-api-authn-webhook.yaml] to node [123.123.123.48]
2022-09-12T07:04:03Z [INFO ] Successfully started [file-deployer] container on host [123.123.123.48]
2022-09-12T07:04:03Z [INFO ] Waiting for [file-deployer] container to exit on host [123.123.123.48]
2022-09-12T07:04:03Z [INFO ] Waiting for [file-deployer] container to exit on host [123.123.123.48]
2022-09-12T07:04:03Z [INFO ] Container [file-deployer] is still running on host [123.123.123.48]: stderr: [], stdout: []
2022-09-12T07:04:04Z [INFO ] Waiting for [file-deployer] container to exit on host [123.123.123.48]
2022-09-12T07:04:04Z [INFO ] [remove/file-deployer] Successfully removed container on host [123.123.123.48]
2022-09-12T07:04:04Z [INFO ] [file-deploy] Deploying file [/etc/kubernetes/kube-api-authn-webhook.yaml] to node [123.123.123.49]
2022-09-12T07:04:04Z [INFO ] Successfully started [file-deployer] container on host [123.123.123.49]
2022-09-12T07:04:04Z [INFO ] Waiting for [file-deployer] container to exit on host [123.123.123.49]
2022-09-12T07:04:04Z [INFO ] Waiting for [file-deployer] container to exit on host [123.123.123.49]
2022-09-12T07:04:04Z [INFO ] Container [file-deployer] is still running on host [123.123.123.49]: stderr: [], stdout: []
2022-09-12T07:04:05Z [INFO ] Waiting for [file-deployer] container to exit on host [123.123.123.49]
2022-09-12T07:04:05Z [INFO ] [remove/file-deployer] Successfully removed container on host [123.123.123.49]
2022-09-12T07:04:05Z [INFO ] [/etc/kubernetes/kube-api-authn-webhook.yaml] Successfully deployed authentication webhook config Cluster nodes
2022-09-12T07:04:05Z [INFO ] [file-deploy] Deploying file [/etc/kubernetes/audit-policy.yaml] to node [123.123.123.48]
2022-09-12T07:04:06Z [INFO ] Successfully started [file-deployer] container on host [123.123.123.48]
2022-09-12T07:04:06Z [INFO ] Waiting for [file-deployer] container to exit on host [123.123.123.48]
2022-09-12T07:04:06Z [INFO ] Waiting for [file-deployer] container to exit on host [123.123.123.48]
2022-09-12T07:04:06Z [INFO ] Container [file-deployer] is still running on host [123.123.123.48]: stderr: [], stdout: []
2022-09-12T07:04:07Z [INFO ] Waiting for [file-deployer] container to exit on host [123.123.123.48]
2022-09-12T07:04:07Z [INFO ] [remove/file-deployer] Successfully removed container on host [123.123.123.48]
2022-09-12T07:04:07Z [INFO ] [file-deploy] Deploying file [/etc/kubernetes/audit-policy.yaml] to node [123.123.123.49]
2022-09-12T07:04:07Z [INFO ] Successfully started [file-deployer] container on host [123.123.123.49]
2022-09-12T07:04:07Z [INFO ] Waiting for [file-deployer] container to exit on host [123.123.123.49]
2022-09-12T07:04:07Z [INFO ] Waiting for [file-deployer] container to exit on host [123.123.123.49]
2022-09-12T07:04:07Z [INFO ] Container [file-deployer] is still running on host [123.123.123.49]: stderr: [], stdout: []
2022-09-12T07:04:08Z [INFO ] Waiting for [file-deployer] container to exit on host [123.123.123.49]
2022-09-12T07:04:08Z [INFO ] [remove/file-deployer] Successfully removed container on host [123.123.123.49]
2022-09-12T07:04:08Z [INFO ] [/etc/kubernetes/audit-policy.yaml] Successfully deployed audit policy file to Cluster control nodes
2022-09-12T07:04:08Z [INFO ] [reconcile] Reconciling cluster state
2022-09-12T07:04:08Z [INFO ] [reconcile] Check etcd hosts to be deleted
2022-09-12T07:04:08Z [INFO ] [reconcile] Check etcd hosts to be added
2022-09-12T07:04:08Z [INFO ] [reconcile] Rebuilding and updating local kube config
2022-09-12T07:04:08Z [INFO ] Successfully Deployed local admin kubeconfig at [management-state/rke/rke-3162177412/kube_config_cluster.yml]
2022-09-12T07:04:08Z [INFO ] [reconcile] host [123.123.123.48] is a control plane node with reachable Kubernetes API endpoint in the cluster
2022-09-12T07:04:08Z [INFO ] [reconcile] Reconciled cluster state successfully
2022-09-12T07:04:08Z [INFO ] Pre-pulling kubernetes images
2022-09-12T07:04:08Z [INFO ] Kubernetes images pulled successfully
2022-09-12T07:04:08Z [INFO ] [etcd] Building up etcd plane..
2022-09-12T07:04:09Z [INFO ] Successfully started [etcd-fix-perm] container on host [123.123.123.48]
2022-09-12T07:04:09Z [INFO ] Waiting for [etcd-fix-perm] container to exit on host [123.123.123.48]
2022-09-12T07:04:09Z [INFO ] Waiting for [etcd-fix-perm] container to exit on host [123.123.123.48]
2022-09-12T07:04:09Z [INFO ] Container [etcd-fix-perm] is still running on host [123.123.123.48]: stderr: [], stdout: []
2022-09-12T07:04:10Z [INFO ] Waiting for [etcd-fix-perm] container to exit on host [123.123.123.48]
2022-09-12T07:04:10Z [INFO ] [remove/etcd-fix-perm] Successfully removed container on host [123.123.123.48]
2022-09-12T07:04:11Z [INFO ] [etcd] Successfully started [rke-log-linker] container on host [123.123.123.48]
2022-09-12T07:04:11Z [INFO ] [remove/rke-log-linker] Successfully removed container on host [123.123.123.48]
2022-09-12T07:04:12Z [INFO ] Successfully started [etcd-fix-perm] container on host [123.123.123.49]
2022-09-12T07:04:12Z [INFO ] Waiting for [etcd-fix-perm] container to exit on host [123.123.123.49]
2022-09-12T07:04:12Z [INFO ] Waiting for [etcd-fix-perm] container to exit on host [123.123.123.49]
2022-09-12T07:04:12Z [INFO ] Container [etcd-fix-perm] is still running on host [123.123.123.49]: stderr: [], stdout: []
2022-09-12T07:04:13Z [INFO ] Waiting for [etcd-fix-perm] container to exit on host [123.123.123.49]
2022-09-12T07:04:13Z [INFO ] [remove/etcd-fix-perm] Successfully removed container on host [123.123.123.49]
2022-09-12T07:04:13Z [INFO ] [etcd] Successfully started [rke-log-linker] container on host [123.123.123.49]
2022-09-12T07:04:14Z [INFO ] [remove/rke-log-linker] Successfully removed container on host [123.123.123.49]
2022-09-12T07:04:14Z [INFO ] [etcd] Successfully started etcd plane.. Checking etcd cluster health
```


And this is the log from the second try where 2 nodes had the error
```
2022-09-12T08:19:42Z [INFO ] Initiating Kubernetes cluster
2022-09-12T08:19:42Z [INFO ] Successfully Deployed state file at [management-state/rke/rke-3423169793/cluster.rkestate]
2022-09-12T08:19:42Z [INFO ] Building Kubernetes cluster
2022-09-12T08:19:42Z [INFO ] [dialer] Setup tunnel for host [123.123.123.49]
2022-09-12T08:19:42Z [INFO ] [dialer] Setup tunnel for host [123.123.123.47]
2022-09-12T08:19:42Z [INFO ] [dialer] Setup tunnel for host [123.123.123.48]
2022-09-12T08:19:42Z [ERROR] Failed to set up SSH tunneling for host [123.123.123.49]: Can't retrieve Docker Info: error during connect: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/info": can not build dialer to [c-7xwnb:m-e52719fe6f35]
2022-09-12T08:19:42Z [ERROR] Failed to set up SSH tunneling for host [123.123.123.48]: Can't retrieve Docker Info: error during connect: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/info": can not build dialer to [c-7xwnb:m-62d047aa1983]
2022-09-12T08:19:42Z [ERROR] Removing host [123.123.123.49] from node lists
2022-09-12T08:19:42Z [ERROR] Removing host [123.123.123.48] from node lists
2022-09-12T08:19:42Z [INFO ] [network] No hosts added existing cluster, skipping port check
2022-09-12T08:19:42Z [INFO ] [certificates] Deploying kubernetes certificates to Cluster nodes
2022-09-12T08:19:48Z [INFO ] [reconcile] Rebuilding and updating local kube config
2022-09-12T08:19:48Z [INFO ] Successfully Deployed local admin kubeconfig at [management-state/rke/rke-3423169793/kube_config_cluster.yml]
2022-09-12T08:19:48Z [INFO ] [reconcile] host [123.123.123.47] is a control plane node with reachable Kubernetes API endpoint in the cluster
2022-09-12T08:19:48Z [INFO ] [certificates] Successfully deployed kubernetes certificates to Cluster nodes
2022-09-12T08:19:48Z [INFO ] [file-deploy] Deploying file [/etc/kubernetes/kube-api-authn-webhook.yaml] to node [123.123.123.47]
2022-09-12T08:19:49Z [INFO ] Successfully started [file-deployer] container on host [123.123.123.47]
2022-09-12T08:19:49Z [INFO ] Waiting for [file-deployer] container to exit on host [123.123.123.47]
2022-09-12T08:19:49Z [INFO ] Waiting for [file-deployer] container to exit on host [123.123.123.47]
2022-09-12T08:19:49Z [INFO ] Container [file-deployer] is still running on host [123.123.123.47]: stderr: [], stdout: []
2022-09-12T08:19:50Z [INFO ] Waiting for [file-deployer] container to exit on host [123.123.123.47]
2022-09-12T08:19:50Z [INFO ] [remove/file-deployer] Successfully removed container on host [123.123.123.47]
2022-09-12T08:19:50Z [INFO ] [/etc/kubernetes/kube-api-authn-webhook.yaml] Successfully deployed authentication webhook config Cluster nodes
2022-09-12T08:19:50Z [INFO ] [file-deploy] Deploying file [/etc/kubernetes/audit-policy.yaml] to node [123.123.123.47]
2022-09-12T08:19:51Z [INFO ] Successfully started [file-deployer] container on host [123.123.123.47]
2022-09-12T08:19:51Z [INFO ] Waiting for [file-deployer] container to exit on host [123.123.123.47]
2022-09-12T08:19:51Z [INFO ] Waiting for [file-deployer] container to exit on host [123.123.123.47]
2022-09-12T08:19:51Z [INFO ] Container [file-deployer] is still running on host [123.123.123.47]: stderr: [], stdout: []
2022-09-12T08:19:52Z [INFO ] Waiting for [file-deployer] container to exit on host [123.123.123.47]
2022-09-12T08:19:52Z [INFO ] [remove/file-deployer] Successfully removed container on host [123.123.123.47]
2022-09-12T08:19:52Z [INFO ] [/etc/kubernetes/audit-policy.yaml] Successfully deployed audit policy file to Cluster control nodes
2022-09-12T08:19:52Z [INFO ] [reconcile] Reconciling cluster state
2022-09-12T08:19:52Z [INFO ] [reconcile] Check etcd hosts to be deleted
2022-09-12T08:19:52Z [INFO ] [reconcile] Check etcd hosts to be added
2022-09-12T08:19:52Z [INFO ] [reconcile] Rebuilding and updating local kube config
2022-09-12T08:19:52Z [INFO ] Successfully Deployed local admin kubeconfig at [management-state/rke/rke-3423169793/kube_config_cluster.yml]
2022-09-12T08:19:52Z [INFO ] [reconcile] host [123.123.123.47] is a control plane node with reachable Kubernetes API endpoint in the cluster
2022-09-12T08:19:52Z [INFO ] [reconcile] Reconciled cluster state successfully
2022-09-12T08:19:52Z [INFO ] Pre-pulling kubernetes images
2022-09-12T08:19:52Z [INFO ] Kubernetes images pulled successfully
2022-09-12T08:19:52Z [INFO ] [etcd] Building up etcd plane..
2022-09-12T08:19:52Z [INFO ] Successfully started [etcd-fix-perm] container on host [123.123.123.47]
2022-09-12T08:19:52Z [INFO ] Waiting for [etcd-fix-perm] container to exit on host [123.123.123.47]
2022-09-12T08:19:52Z [INFO ] Waiting for [etcd-fix-perm] container to exit on host [123.123.123.47]
2022-09-12T08:19:52Z [INFO ] Container [etcd-fix-perm] is still running on host [123.123.123.47]: stderr: [], stdout: []
2022-09-12T08:19:53Z [INFO ] Waiting for [etcd-fix-perm] container to exit on host [123.123.123.47]
2022-09-12T08:19:53Z [INFO ] [remove/etcd-fix-perm] Successfully removed container on host [123.123.123.47]
2022-09-12T08:19:54Z [INFO ] [etcd] Successfully started [rke-log-linker] container on host [123.123.123.47]
2022-09-12T08:19:54Z [INFO ] [remove/rke-log-linker] Successfully removed container on host [123.123.123.47]
2022-09-12T08:19:54Z [INFO ] [etcd] Successfully started etcd plane.. Checking etcd cluster health
```


And this is the cluster-config:

```
answers: {}
description: For some stuff
docker_root_dir: /var/lib/docker
enable_cluster_alerting: false
enable_cluster_monitoring: false
enable_network_policy: false
fleet_workspace_name: fleet-default
local_cluster_auth_endpoint:
  ca_certs: |-
    -----BEGIN CERTIFICATE-----
    BLOOPBLAP-BASE64-STUFF
    -----END CERTIFICATE-----
  enabled: true
  fqdn: some.k8s-api.local.domain
name: our-cluster
rancher_kubernetes_engine_config:
  addon_job_timeout: 30
  authentication:
    strategy: x509|webhook
  authorization: {}
  bastion_host:
    ignore_proxy_env_vars: false
    ssh_agent_auth: false
  cloud_provider: {}
  ignore_docker_version: true
  ingress:
    default_backend: true
    default_ingress_class: true
    http_port: 0
    https_port: 0
    provider: nginx
  kubernetes_version: v1.22.10-rancher1-1
  monitoring:
    provider: metrics-server
    replicas: 1
  network:
    mtu: 0
    options:
      flannel_backend_type: vxlan
    plugin: canal
  restore:
    restore: false
  rotate_encryption_key: false
  services:
    etcd:
      backup_config:
        enabled: true
        interval_hours: 12
        retention: 6
        safe_timestamp: false
        timeout: 300
      creation: 12h
      extra_args:
        election-timeout: '5000'
        heartbeat-interval: '500'
      gid: 0
      retention: 72h
      snapshot: false
      uid: 0
    kube-api:
      always_pull_images: false
      audit_log:
        configuration:
          format: json
          max_age: 3
          max_backup: 3
          max_size: 100
          path: /var/log/kube-audit/audit-log.json
          policy:
            api_version: audit.k8s.io/v1
            kind: Policy
            metadata: {}
            omit_stages:
              - RequestReceived
            rules:
              - level: RequestResponse
                resources:
                  - resources:
                      - configmaps
                      - daemonsets
                      - deployments
                      - ingressclasses
                      - ingresses
                      - namespaces
                      - nodes
                      - persistentvolumes
                      - persistentvolumeclaims
                      - pods
                      - replicasets
                      - secrets
                      - services
                      - statefulsets
                      - storageclasses
                verbs:
                  - create
                  - delete
                  - patch
                  - update
        enabled: true
      pod_security_policy: false
      service_node_port_range: 30000-32767
    kube-controller: {}
    kubelet:
      fail_swap_on: false
      generate_serving_certificate: false
    kubeproxy: {}
    scheduler: {}
  ssh_agent_auth: false
  upgrade_strategy:
    drain: false
    max_unavailable_controlplane: '1'
    max_unavailable_worker: 10%
    node_drain_input:
      delete_local_data: false
      force: false
      grace_period: -1
      ignore_daemon_sets: true
      timeout: 120
scheduled_cluster_scan:
  enabled: false
  scan_config:
    cis_scan_config:
      override_benchmark_version: rke-cis-1.5
      profile: permissive
  schedule_config:
    cron_schedule: 0 0 * * *
    retention: 24
```

**Comment by markusewalker** [comment]
@BBQigniter at least from my inspection of your logs above, this seems to be a different issue from the originally reported issue. @kinarashah can you please confirm if this is the case. If so, it would be best to open up a separate issue from this.

I am in the midst of validating the original issue reported and, if successful, will be closing this issue.

**Comment by kinarashah** [comment]
Fix merged in rke `v1.4.0-rc1` and has been vendored into Rancher.

**Comment by markusewalker** [comment]
Validated that this is addressed in `v2.7-head` https://github.com/rancher/rancher/commit/c567ef3f2bd13c2f2a5819242ab88cd334b1ae79.

**ENVIRONMENT DETAILS**
- Rancher install: Docker
- Rancher version:
     - Fresh install: `v2.7-head` https://github.com/rancher/rancher/commit/c567ef3f2bd13c2f2a5819242ab88cd334b1ae79
     - Upgrade: `v2.6.8` -> `v2.7-head` https://github.com/rancher/rancher/commit/c567ef3f2bd13c2f2a5819242ab88cd334b1ae79
- Browser: Chrome

**TEST RESULT**
Scenario 1: Rancher Fresh Install - PASS
Scenario 2: Rancher Upgrade - PASS
Scenario 3: RKE Standalone - PASS

**VERIFICATION STEPS**
Scenario 1: Fresh install Rancher v2.7-head
1. Setup Rancher, created a standard user and logged in as that user.
2. Provisioned a downstream RKE1 node driver cluster with K8s `v1.23.10`.
     - Edited the cluster's YAML before cluster creation to have the following addons underneath the `rancher_kubernetes_engine_config` block:
```
addon_job_timeout: 30
addons: |-
  ---
  apiVersion: v1
  kind: Namespace
  metadata:
    name: cattle-system
```
3. Validated that the cluster and nodes came up as `Active` in Cluster Management and Cluster Explorer.
4. Edited the cluster's config, validated that the addons still appear:
![image](https://user-images.githubusercontent.com/52591819/192611623-50d1f90a-5d2d-4c64-87b6-3577252cf163.png) 
5. Upgraded the cluster's K8s version to `v1.24.4`;  validated that this was successful:
![image](https://user-images.githubusercontent.com/52591819/192611641-730f2ed8-e173-4a6f-8eeb-839c40705191.png)
______
Scenario 2: Upgrade Rancher from v2.6.8 -> v2.7-head
1. Setup Rancher `v2.6.8`, create a standard user and login as that user.
2. Provisioned a downstream RKE1 node driver cluster for the highest K8s `v1.23.x` version.
     - While provisioning, edited the cluster's YAML and added the following underneath the `rancher_kubernetes_engine_config` block:
```
addon_job_timeout: 30
addons: |-
  ---
  apiVersion: v1
  kind: Namespace
  metadata:
    name: cattle-system
```
3. Validated that the cluster and nodes appear in a state of `Active` in Cluster Management and Cluster Explorer.
4. Edited the cluster's YAML and validated that the addons have been preserved:
![image](https://user-images.githubusercontent.com/52591819/192611778-166ad7f5-f22b-42f7-823e-0525b14f78eb.png) 
5. Upgraded Rancher from `v2.6.8` to `v2.7-head`.
6. Validated that the cluster did not auto-upgrade to the next K8s version and that it still is in a state of `Active`.
7. Edited the cluster's YAML and validated that the addons have been preserved.
     - Updated the K8s version.
8. Validated that the cluster and nodes have been upgraded to the latest K8s version and the state `Active` is in Cluster Explorer and Cluster Management.
______
Scenario 3: Standalone RKE1
1. On a client machine, setup RKE version `v1.4.0-rc1`.
2. In the `cluster.yml`, added the following addon:

<img width="588" alt="image" src="https://user-images.githubusercontent.com/52591819/192611863-398a2a83-c481-4c02-af04-39f199ae093a.png"> 

3.. Ran `rke up --config cluster.yml` and validated that the cluster came up successfully:
```
$ kubectl get nodes
NAME       STATUS   ROLES                      AGE   VERSION
mew-rke1   Ready    controlplane,etcd,worker   65s   v1.24.4
mew-rke2   Ready    controlplane,etcd,worker   65s   v1.24.4
mew-rke3   Ready    controlplane,etcd,worker   64s   v1.24.4
$ kubectl get pods -A
NAMESPACE       NAME                                       READY   STATUS      RESTARTS   AGE
ingress-nginx   ingress-nginx-admission-create-96ms4       0/1     Completed   0          30s
ingress-nginx   ingress-nginx-admission-patch-sd4gl        0/1     Completed   1          30s
ingress-nginx   nginx-ingress-controller-7bsjg             1/1     Running     0          30s
ingress-nginx   nginx-ingress-controller-jmdwp             1/1     Running     0          30s
ingress-nginx   nginx-ingress-controller-n879f             1/1     Running     0          30s
kube-system     calico-kube-controllers-74df54cbb7-kmtg4   1/1     Running     0          60s
kube-system     canal-25rbg                                2/2     Running     0          60s
kube-system     canal-gbmtt                                2/2     Running     0          60s
kube-system     canal-shdrb                                2/2     Running     0          60s
kube-system     coredns-59499769fb-8bpbq                   1/1     Running     0          37s
kube-system     coredns-59499769fb-n8n6n                   1/1     Running     0          50s
kube-system     coredns-autoscaler-67cbd4599c-vwbfx        1/1     Running     0          50s
kube-system     metrics-server-585b7cc746-jrwzs            1/1     Running     0          40s
kube-system     rke-coredns-addon-deploy-job-t792n         0/1     Completed   0          52s
kube-system     rke-ingress-controller-deploy-job-kb5rc    0/1     Completed   0          32s
kube-system     rke-metrics-addon-deploy-job-nspf2         0/1     Completed   0          42s
kube-system     rke-network-plugin-deploy-job-5ss99        0/1     Completed   0          62s
kube-system     rke-user-addon-deploy-job-cl4wh            0/1     Completed   0          22s
```
4. Re-edited the cluster.yml and validated that the addons remained.
5. Re-ran `rke up --config cluster.yml` with no changes; validated that the `rke-user-addon` got recreated.

**Comment by Tejeev** [comment]
Some actions that had to be taken after running into this bug in order to get the cluster back online (in case it helps anyone out there):

After a VM restore, there were a handful of issues that were worked through:
- cattle-cluster-agent and ingress-nginx namespaces being deleted. Removed an old rke-user-addon configmap to stop this from happening and then applied the clusterregistrationtoken to re-create these workloads.
- After getting these workloads up and running, they still wouldn't connect back to Rancher, so they triggered the kube-api-auth job to run by toggling the authorizedClusterEndpoint in the cluster spec from Enabled to Disabled and back.
- Also recreated the ingress-nginx workloads by toggling the ingress.provider from "nginx" to "none" and back. This triggers the ingress-nginx-deploy job in rke, which created the missing resources.

From here, the cluster came up and was mostly healthy. There were a couple of workloads that were unhealthy, but this was because they were leftover from the vm snapshot restore. They cleared these out by restarting docker on all of their nodes.
